### PR TITLE
test(move-file): Added test to renaming fallback

### PR DESCRIPTION
**Pull request by ruyadorno**
- Added test that covers the untested renaming fallback using the move-concurrently module, it uses `requireInject` in order to mock `fs`
calls and trigger the correct steps necessary
- Added an istanbul ignore line in order to alleviate the burden on testing very specific platform-dependent errors
- Bumps `lib/util/move-file.js` module to 100% coverage

### CODE REVIEW

File: lib/util/move-file.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ function moveFile (src, dest) {
   return new Promise((resolve, reject) => {
     fs.link(src, dest, (err) => {
       if (err) {
+        /* istanbul ignore else */
```
Comment (isaacs) [inline lib/util/move-file.js]: I think this should be testable with http://npm.im/mutate-fs

Comment (ruyadorno) [reply lib/util/move-file.js]: oh, almost all the lines here already have coverage! the only missing line is **L29** since it's windows specific, maybe I should move the comment to the same line 🤔

Comment (ruyadorno) [reply lib/util/move-file.js]: ok that should be much better now, I moved it to stay in the same line of that platform-dependent condition and it only ignores that single statement instead of all `else` blocks 😊

File: lib/util/move-file.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ function moveFile (src, dest) {
       if (err) {
         if (err.code === 'EEXIST' || err.code === 'EBUSY') {
           // file already exists, so whatever
-        } else if (err.code === 'EPERM' && process.platform === 'win32') {
+        }/* istanbul ignore next */ else if (err.code === 'EPERM' && process.platform === 'win32') {
```
Comment (isaacs) [inline lib/util/move-file.js]: This should be `/* istanbul ignore if */` so that we still cover the `else` block below it.

Comment (ruyadorno) [reply lib/util/move-file.js]: that was my first intuition too but then it turns out using `/* istanbul ignore if */` does not really works as I was expecting (**L28** still misses coverage, same as not having the istanbul comment at all)

that's how I ended up moving it to the **L26** and using `/* istanbul ignore else */` instead (the original commit I had in this PR) 😞

Comment (isaacs) [reply lib/util/move-file.js]: (posted the bug to istanbuljs here: https://github.com/istanbuljs/istanbuljs/issues/477)

File: lib/util/move-file.js [hunk 3]
```diff
@@ -25,9 +25,11 @@ function moveFile (src, dest) {
       if (err) {
         if (err.code === 'EEXIST' || err.code === 'EBUSY') {
           // file already exists, so whatever
-        } else if (err.code === 'EPERM' && process.platform === 'win32') {
-          // file handle stayed open even past graceful-fs limits
         } else {
+          /* istanbul ignore next */
+          if (err.code === 'EPERM' && process.platform === 'win32') {
+            // file handle stayed open even past graceful-fs limits
+          }
```
Comment (ruyadorno) [inline lib/util/move-file.js]: @isaacs I couldn't get the `/* istanbul ignore if */` to work as the [example you set up](https://gist.github.com/isaacs/987fc367e2422ba782de9298586ea5c7#file-ignore-if-works-js) 😵  I wonder if it's the version of `nyc` we're running or if I'm messing up with something 🤔 

it did worked now with the `/* istanbul ignore next */` and in my understanding that's fine now since there's no `else` as a part of that same block

Comment (isaacs) [reply lib/util/move-file.js]: So, just a pedantic note, these are not the same thing:

```js
/* istanbul ignore next */
if (conditionNeverMet) {
  foo()
}

/* istanbul ignore if */
if (conditionNeverMet) {
  foo()
}
```

The first ignores both the if _and_ else paths.  The second ignores only the `if` branch, but verifies that the `else` branch was hit.

They appear the same in this case because the `else` path _is_ getting hit.  But if at some point that changed, and instead of never being satisfied, it was _always_ being satisfied, then the `/* istanbul ignore if */` would show missing branch coverage, and `/* istanbul ignore next */` would not.

In other words, from istanbul's point of view _every_ `if` has an `else`, even if there's no functional code there.  Imagine that it sees the code like this:

```js
if (condition) {
  doSomething()
} else {
  doNothing()
}
```

and then it's verifying that the `doNothing` branch is being traversed.

In general, it's best to ignore as little as possible.

Comment (ruyadorno) [reply lib/util/move-file.js]: thanks for the note! that confirms my understanding of it 👍 

I can try to get down to the bottom of the problem later 😊
